ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh

```diff
@@ -336,6 +357,113 @@ function upgrade() {
+                exit 1
+            fi
+        fi
+        # first_upgrade_to_image can be both nightly build or stable build
+        run_command "oc adm upgrade --to-image=${first_upgrade_to_image} --allow-explicit-upgrade --force=${FORCE_UPDATE}"
+        wait_upgrade_start "Error: OCP-25473: First time upgrade not started, break the job"
+
+        if [[ "$MINOR_VERSION_FOR_TEST" == "$TARGET_MINOR_VERSION" ]]; then
+            oc -n openshift-config-managed patch configmap admin-gates \
+                --type json \
+                -p '[{"op": "add", "path": "/data", "value": {"ack-4.'${MINOR_VERSION_FOR_TEST}'-testing": "gate-testing"}}]'
+            sleep 1m
+            output="$(oc adm upgrade)"
+            if ! [[ "$output" =~ "gate-testing" ]] \
+                || ! [[ "$output" =~ "Upgradeable=False" ]]; then
+                echo "Error: OCP-25473: After patch admin-gates, the message is not correct, the observed output is:"
+                echo $output
+                exit 1
+            fi
+        fi
+
+        echo "OCP-25473: Upgrade to new target without --allow-upgrade-with-warnings"
+        # second_upgrade_to_image must be ***stable*** build
+        output="$(oc adm upgrade --to-image=${second_upgrade_to_image} --allow-explicit-upgrade 2>&1 || true)"
+        if [[ ! "${output}" =~ "error: the cluster is already upgrading" ]] || [[ ! "${output}" =~ "If you want to upgrade anyway, use --allow-upgrade-with-warnings" ]]; then
+            echo "Error: OCP-25473: Re-upgrade should not started and raise error when there is already an upgrade is processing"
+            echo "The output of 'oc adm upgrade' is:"
+            echo $output
+            exit 1
+        fi
+        
+        echo "OCP-25473: Upgrade to new target again with --allow-upgrade-with-warnings"
+        run_command "oc adm upgrade --to-image=${second_upgrade_to_image} --allow-explicit-upgrade --allow-upgrade-with-warnings=true"
+        sleep 1m
+        if [[ "$MINOR_VERSION_FOR_TEST" -gt "$TARGET_MINOR_VERSION" ]]; then
+            echo "This is re-targeting to a y version, second upgrade should be blocked"
+            output="$(oc get clusterversion version -ojson)"
+            if [[ "$(echo $output | jq -r '.status.conditions[] | select(.type == "Progressing").status')" != "True" ]] \
+                || [[ "$(echo $output | jq -r '.status.conditions[] | select(.type == "Upgradeable").status')" != "False" ]] \
+                || [[ "$(echo $output | jq -r '.status.conditions[] | select(.type == "ReleaseAccepted").status')" != "False" ]] \
+                || ! [[ "$(echo $output | jq -r '.status.conditions[] | select(.type == "ReleaseAccepted").message')" =~ ${second_upgrade_to_image} ]]; then
+                echo "Error: OCP-25473: Retarget to a y stream version should be blocked, but actually not"
+                exit 1
+            fi
+            run_command "oc adm upgrade --clear"
+            sleep 1m
```

Comment:
this is a good idea! we want to clear so once the previous update is completed, we don't accidentally unblock. as for the sleep, I don't recall how much time it actually takes, but usually its ~20seconds for changes like this setting a new upgrade target, so 1m should be enough. btw there is some other kind of changes that happens in the cluster, for which cvo will have a reconciliation loop that runs every 2-4 minutes (this interval is set at random between those two, at the moment cvo starts the first time..) but if i recall, the effect of --clear is not this other kind.